src/resource/resource.ts

```diff
@@ -338,6 +339,27 @@ export function isPodOwner(
   return podOwner === webId;
 }
 
+/**
+ * Extends the regular JavaScript error object with access to the status code and status message.
+ */
+export class FetchError extends Error {
```

Comment:
I'd have expected this class to be in it's own source file, i.e. `FetchError.ts`.

Reply:
Just because it's a class? We don't use separate files for all the functions related to Resources either?

Reply:
No, not **_just_** because it's a class, but because it seems to be a very general, generic, standalone bit of code that I assume could be easily reused across much of this codebase, or in fact any Inrupt codebase. Having it reside within the `resource.ts` source file would, I think, makes any developer think it is somehow be related to 'resources'. In fact, looking at the `resource.ts` file itself now, it seems to be a kinda grab-bag of utility functions related to resources (so I'd probably rename it `resourceUtil.ts`, if that is it's actual intent), but that seems to reinforce my opinion of moving this Fetch-specific Error class into it's own source file, as I wouldn't see it merely as a util class for use with resources, I'd see it as a general-purpose Error class for use with fetch, any fetch, and therefore seeing it reside in a file named `FetchError.ts` would be my expectation.

Reply:
Currently, the organisation is as follows:

```
src
├─  resource/ # Contains everything related to fetching data
| ├─ solidDataset.ts # Specific to fetching SolidDatasets
| ├─ nonRdfData.ts # Specific to fetching non-RDF data
| ├─ resource.ts # Generic data fetching code
```

By that organisation, it seems logical to have errors arising from fetching data of any kind inside `/resource/resource.ts`?

Reply:
I guess it's subjective - i.e. it's not really a big deal (so I'm happy to leave this one), but personally I'd rename `resource.ts` to `resourceUtil.js` (as that seems to better reflect it's intent), and provide the `FetchError` class in either the existing `/fetcher.ts` file, or it's own file in a project `Util` folder (in fact, perhaps even a generic Inrupt JavaScript Util project (but I'm not sure if we have enough common utility JavaScript code across the various Inrupt teams yet to justify that step, but consistent error handling would be a very good candidate for starting...)).

Reply:
Haha OK, that's indeed my definitely subjective and unreasonably strong dislike of anything called "util" or that is the kind of grab-bag of hardly-related too-generic functionality that it usually refers to, which I will resist for longer than a reasonable person would consider worth it :P

(But of course, if this does turn out to be something that we use beyond the resource-related functionality, we can always move it elsewhere.)